src/cart/crt_bulk.c

```diff
@@ -231,6 +231,9 @@ crt_bulk_free(crt_bulk_t crt_bulk)
 	/* decoded bulks are not counted towards quota; such bulks have crt_ctx set to NULL */
 	if (bulk->crt_ctx)
 		put_quota_resource(bulk->crt_ctx, CRT_QUOTA_BULKS);
+
+	if (bulk->bulk_orig_uri)
+		D_FREE(bulk->bulk_orig_uri);
```

Comment:
this should be at the out label where the rest of bulk is free-ed

Reply:
ah, yes. thanks